x-pack/plugins/apm/server/lib/search_strategies/failed_transactions_correlations/async_search_service.ts

```diff
@@ -5,32 +5,58 @@
 import { chunk } from 'lodash';
-import type { SearchServiceParams } from '../../../../common/search_strategies/correlations/types';
-import type { ApmIndicesConfig } from '../../settings/apm_indices/get_apm_indices';
-import { asyncSearchServiceLogProvider } from '../correlations/async_search_service_log';
-import { asyncErrorCorrelationsSearchServiceStateProvider } from './async_search_service_state';
-import { fetchTransactionDurationFieldCandidates } from '../correlations/queries';
-import type { SearchServiceFetchParams } from '../../../../common/search_strategies/correlations/types';
-import { fetchFailedTransactionsCorrelationPValues } from './queries/query_failure_correlation';
-import { ERROR_CORRELATION_THRESHOLD } from './constants';
+
+import type { ElasticsearchClient } from 'src/core/server';
+
+import type { ISearchStrategy } from '../../../../../../../src/plugins/data/server';
+import {
+  IKibanaSearchRequest,
+  IKibanaSearchResponse,
+} from '../../../../../../../src/plugins/data/common';
+
 import { EVENT_OUTCOME } from '../../../../common/elasticsearch_fieldnames';
+import type { SearchStrategyParams } from '../../../../common/search_strategies/types';
+import type {
+  FailedTransactionsCorrelationsParams,
+  FailedTransactionsCorrelationsRawResponse,
+} from '../../../../common/search_strategies/failed_transactions_correlations/types';
+import type { ApmIndicesConfig } from '../../settings/apm_indices/get_apm_indices';
+import { asyncSearchServiceLogProvider } from '../async_search_service_log';
+import {
+  fetchFailedTransactionsCorrelationPValues,
+  fetchTransactionDurationFieldCandidates,
+} from '../queries';
+import type { AsyncSearchServiceProvider } from '../search_strategy_provider';
+
+import { asyncFailedTransactionsCorrelationsSearchServiceStateProvider } from './async_search_service_state';
 
-export const asyncErrorCorrelationSearchServiceProvider = (
+import { ERROR_CORRELATION_THRESHOLD } from '../constants';
+
+export type FailedTransactionsCorrelationsAsyncSearchServiceProvider = AsyncSearchServiceProvider<
+  FailedTransactionsCorrelationsParams,
+  FailedTransactionsCorrelationsRawResponse
+>;
+
+export type FailedTransactionsCorrelationsSearchStrategy = ISearchStrategy<
+  IKibanaSearchRequest<FailedTransactionsCorrelationsParams>,
+  IKibanaSearchResponse<FailedTransactionsCorrelationsRawResponse>
+>;
+
+export const failedTransactionsCorrelationsAsyncSearchServiceProvider: FailedTransactionsCorrelationsAsyncSearchServiceProvider = (
   esClient: ElasticsearchClient,
   getApmIndices: () => Promise<ApmIndicesConfig>,
-  searchServiceParams: SearchServiceParams,
+  searchServiceParams: FailedTransactionsCorrelationsParams,
   includeFrozen: boolean
 ) => {
   const { addLogMessage, getLogMessages } = asyncSearchServiceLogProvider();
 
-  const state = asyncErrorCorrelationsSearchServiceStateProvider();
+  const state = asyncFailedTransactionsCorrelationsSearchServiceStateProvider();
```

Comment:
Is the `async` part of the name needed here? SearchStrategy (= Kibana naming) + SearchService (our internal process to do the searches) should be enough. If it is needed, should the `async` part of the name is not consistent with e.g. `latencyCorrelationsAsyncSearchServiceProvider`

Reply:
Fixed inconsistencies in 1656d8d.